wsd/Storage.cpp

```diff
@@ -739,6 +739,17 @@ WopiStorage::WOPIFileInfo::WOPIFileInfo(const FileInfo &fileInfo,
     JsonUtil::findJSONValue(object, "TemplateSaveAs", _templateSaveAs);
     JsonUtil::findJSONValue(object, "TemplateSource", _templateSource);
 
+    // UserFriendlyName is used as the Author when loading the document.
+    // If it's missing, document loading fails. Since the UserFriendlyName
+    // field is optional in WOPI specs, it's often left out by integrators.
+    if (_username.empty())
+    {
+        _username = "Unknown"; // Default to something sensible yet friendly.
```

Comment:
Why not to default to _userId, and only to "Unknown" when even that is empty (?)

Reply:
> Why not to default to _userId, and only to "Unknown" when even that is empty (?)

Isn't _userId almost always an integer? At least that what I noticed. If so, the Author name would show up as a number, which is probably more confusing than Unknown. (I actually thought of the same idea too.)

Reply:
WOPI defines it as a string, I think the fact that it is a number in richdocuments is an implementation detail, and it already implements the UserFriendlyName anyway.

But up to you really, I don't insist :-)

Reply:
> WOPI defines it as a string, I think the fact that it is a number in richdocuments is an implementation detail, and it already implements the UserFriendlyName anyway.

Yes, I meant the contents are typically some number (possibly a DB foreign key or something).

Anyway, I came up with a compromise: UnknownUser_<UserId>

I expect this will be both unique and meaningful. My concern with just UserId is that it might look like a bug on our side (all the users show as some random numbers), rather than missing information on the WOPI side.

Another minor concern is that UserId *might* be thought to be somewhat sensitive information (I've heard this opinion at least once). But this is temporary, and shouldn't get to production (although it will be used *if* the username field is optional and left blank).

Anyway, let me know if you think my suggestion is sensible or not. I don't insist on any solution either, just trying to find a good balance.

Thanks!